engine/src/test/java/org/hibernate/validator/test/cfg/ProgrammaticConstraintDefinitionsTest.java

```diff
@@ -61,6 +63,19 @@ public void safeHtmlProgrammaticDefinition() {
 		doProgrammaticTest( new SafeHtmlDef().whitelistType( SafeHtml.WhiteListType.NONE ), "test", 0 );
 		doProgrammaticTest( new SafeHtmlDef().whitelistType( SafeHtml.WhiteListType.RELAXED ), "<td>1234qwer</td>", 0 );
 		doProgrammaticTest( new SafeHtmlDef().whitelistType( SafeHtml.WhiteListType.NONE ).additionalTags( "td" ), "<td>1234qwer</td>", 0 );
+
+		AnnotationDescriptor<SafeHtml.Tag> tagDescriptor = new AnnotationDescriptor( SafeHtml.Tag.class );
+		tagDescriptor.setValue( "name", "td" );
+		tagDescriptor.setValue( "attributes", new String[]{ "class", "id" } );
+		doProgrammaticTest( new SafeHtmlDef().whitelistType( SafeHtml.WhiteListType.NONE )
+				.additionalTagsWithAttributes( AnnotationFactory.create( tagDescriptor ) ), "<td class='class' id='tableId'>1234qwer</td>", 0 );
+
+		AnnotationDescriptor<SafeHtml.Tag> protocolDescriptor = new AnnotationDescriptor( SafeHtml.Tag.class );
+		protocolDescriptor.setValue( "name", "img" );
+		protocolDescriptor.setValue( "attributes", new String[]{ "src" } );
+		protocolDescriptor.setValue( "protocols", new String[]{ "data" } );
+		doProgrammaticTest( new SafeHtmlDef().whitelistType( SafeHtml.WhiteListType.NONE )
```

Comment:
yes of course :) 
So we should deprecate current `SafeHtmlDef#additionalTagsWithAttributes` and add something like:
```java
public SafeHtmlDef additionalTagWithAttributes(String name, String[] attributes, String[] protocols) {
		// ...
		return this;
	}
```
or
```java
	public SafeHtmlDef additionalTagsWithAttributes(Tag... additionalTagsWithAttributes) {
		// ...
		return this;
	}

	public static class Tag {
		private final String name;

		private final String[] attributes;

		private final String[] protocols;

		public Tag(String name, String[] attributes, String[] protocols) {
			this.name = name;
			this.attributes = attributes;
			this.protocols = protocols;
		}
	}
```
or
```java
public Tag addTag(String tagName){
		return new Tag(tagName); 
	}

	public class Tag {
		private final String name;

		private final Set<String> attributes;

		private final Set<String> protocols;

		private Tag(String name) {
			this.name = name;
			this.attributes = CollectionHelper.newHashSet();
			this.protocols = CollectionHelper.newHashSet();
		}

		public Tag addAttribute(String name) {
			attributes.add( name );
			return this;
		}

		public Tag addProtocol(String name) {
			protocols.add( name );
			return this;
		}
		
		public SafeHtmlDef add(){
			//...
			return SafeHtmlDef.this;
		}
	}
```
instead, right ?

@gunnarmorling which option would you prefer ?